packages/amplify-graphql-transformer-core/src/utils/model-datasource-strategy-utils.ts

```diff
@@ -60,6 +61,21 @@ export const getModelDataSourceStrategy = (ctx: DataSourceStrategiesProvider, ty
   return strategy;
 };
 
+/**
+ * Type predicate that returns true if `obj` is a AmplifyDynamoDbModelDataSourceStrategy
+ */
+export const isImportedAmplifyDynamoDbModelDataSourceStrategy = (
+  strategy: ModelDataSourceStrategy,
+): strategy is ImportedAmplifyDynamoDbModelDataSourceStrategy => {
+  return (
+    isDynamoDbType(strategy.dbType) &&
+    typeof (strategy as any)['provisionStrategy'] === 'string' &&
+    (strategy as any)['provisionStrategy'] === 'IMPORTED_AMPLIFY_TABLE' &&
+    typeof (strategy as any)['tableName'] === 'string' &&
+    (strategy as any)['tableName'] !== ''
```

Comment:
Should we turn this "empty imported table name" scenario handling into a validation check and fail instead of treating this as a greenfield scenario?

Reply:
This would actually fail with `No resource generator assigned for Post with dbType DYNAMODB`. The error message could be improved though. I can tackle this in the next PR.